From 8d14abb0df1e216682fa9c35bda3d4ecdb9b4dd2 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Fri, 1 Nov 2013 16:45:52 +0000
Subject: [PATCH 1/2] builder: Allow checksum= as an alias for
 checksum[sha512]=.

This is for backwards compatibility with virt-builder 1.24.0 which
could not parse a key that contains square brackets and numbers.

This updates commit 4b73e0be29302da0ae8c5c79a51e770cbe6ed7cf.
---
 builder/index_parser.ml   |  6 ++++--
 builder/virt-builder.pod  |  9 +++++++++
 builder/website/index     | 16 ++++++++--------
 builder/website/index.asc | 42 +++++++++++++++++++++---------------------
 4 files changed, 42 insertions(+), 31 deletions(-)

diff --git a/builder/index_parser.ml b/builder/index_parser.ml
index b9d706b..a8c427c 100644
--- a/builder/index_parser.ml
+++ b/builder/index_parser.ml
@@ -215,7 +215,7 @@ let get_index ~debug ~downloader ~sigchecker source =
               | "osinfo" -> false
               | "file" -> false
               | "sig" -> false
-              | "checksum[sha512]" -> false
+              | "checksum" | "checksum[sha512]" -> false
               | "revision" -> false
               | "format" -> false
               | "size" -> false
@@ -290,7 +290,9 @@ let get_index ~debug ~downloader ~sigchecker source =
             with Not_found -> None in
           let checksum_sha512 =
             try Some (List.assoc "checksum[sha512]" fields)
-            with Not_found -> None in
+            with Not_found ->
+              try Some (List.assoc "checksum" fields)
+              with Not_found -> None in
           let revision =
             try int_of_string (List.assoc "revision" fields)
             with
diff --git a/builder/virt-builder.pod b/builder/virt-builder.pod
index 53b7817..4e33bc8 100644
--- a/builder/virt-builder.pod
+++ b/builder/virt-builder.pod
@@ -1087,6 +1087,15 @@ downloaded.  To work out the signature, do:
 Note if you use this, you don't need to sign the file, ie. don't use
 C<sig>.  This option overrides C<sig>.
 
+=item C<checksum=7b882fe9b82eb0fef...>
+
+C<checksum> is an alias for C<checksum[sha512]>.
+
+If you need to interoperate with virt-builder = 1.24.0 then you have
+to use C<checksum> because that version would give a parse error with
+square brackets and numbers in the key of a field.  This is fixed in
+virt-builder E<ge> 1.24.1.
+
 =item C<revision=N>
 
 The revision is an integer which is used to control the template
diff --git a/builder/website/index b/builder/website/index
index c34934c..6165f28 100644
--- a/builder/website/index
+++ b/builder/website/index
@@ -4,7 +4,7 @@ revision=4
 osinfo=centos6.4
 file=centos-6.xz
 sig=centos-6.xz.sig
-checksum[sha512]=cc961bee19e0ed9573f2fa1bb26923bc77b882fe9b82eb0fefdf969237b586ebb6b4a02818321ace6cc3604d8f0f60e76f22039bf301093fa944c27399f70904
+checksum=cc961bee19e0ed9573f2fa1bb26923bc77b882fe9b82eb0fefdf969237b586ebb6b4a02818321ace6cc3604d8f0f60e76f22039bf301093fa944c27399f70904
 format=raw
 size=6442450944
 compressed_size=191253016
@@ -26,7 +26,7 @@ name=Debian 6 (Squeeze)
 osinfo=debian6
 file=debian-6.xz
 sig=debian-6.xz.sig
-checksum[sha512]=1baff51aae7cb89d96c5a95bcd18f3f9ce04945366802107a4ed3b5bd3f9c1fa77af8aa9716cea2826211ef03855a3f02dfa2097daeae66d4ec20a348c820354
+checksum=1baff51aae7cb89d96c5a95bcd18f3f9ce04945366802107a4ed3b5bd3f9c1fa77af8aa9716cea2826211ef03855a3f02dfa2097daeae66d4ec20a348c820354
 format=raw
 size=4294967296
 compressed_size=141531780
@@ -56,7 +56,7 @@ name=Debian 7 (Wheezy)
 osinfo=debian7
 file=debian-7.xz
 sig=debian-7.xz.sig
-checksum[sha512]=1c36eb94401acc567a1c542e4eafde2ba15dab6b615eb5a5bf4e18895c30d825e2e57c6d3a62f998dfc232f3747d5c92d2c5a497ac5d62d73426da5bf80652a5
+checksum=1c36eb94401acc567a1c542e4eafde2ba15dab6b615eb5a5bf4e18895c30d825e2e57c6d3a62f998dfc232f3747d5c92d2c5a497ac5d62d73426da5bf80652a5
 format=raw
 size=4294967296
 compressed_size=149230076
@@ -86,7 +86,7 @@ name=Fedora速 18
 osinfo=fedora18
 file=fedora-18.xz
 sig=fedora-18.xz.sig
-checksum[sha512]=12435775193b69f6e22658aaa001d4ca9b15fd68a04b4b7e9be20b3b517e857e417dc3268a302979d4a702b20f25754025f7ae0e9fb7088419a4ca1669585e6f
+checksum=12435775193b69f6e22658aaa001d4ca9b15fd68a04b4b7e9be20b3b517e857e417dc3268a302979d4a702b20f25754025f7ae0e9fb7088419a4ca1669585e6f
 format=raw
 size=6442450944
 compressed_size=148947524
@@ -108,7 +108,7 @@ name=Fedora速 19
 osinfo=fedora19
 file=fedora-19.xz
 sig=fedora-19.xz.sig
-checksum[sha512]=6228792b15df695260eea5530770e22b1bce40a1986410252928adbe4131ab85c031bcf8892736b63240fd585e39a5f44ed7770ba0dc1e6d8de6834cccee3767
+checksum=6228792b15df695260eea5530770e22b1bce40a1986410252928adbe4131ab85c031bcf8892736b63240fd585e39a5f44ed7770ba0dc1e6d8de6834cccee3767
 format=raw
 size=4294967296
 compressed_size=172190964
@@ -130,7 +130,7 @@ name=Ubuntu 10.04 (Lucid)
 osinfo=ubuntulucid
 file=ubuntu-10.04.xz
 sig=ubuntu-10.04.xz.sig
-checksum[sha512]=9e322782bc77c740ce956d57d0d89ec9ac9f2b890c474ac937f400db848df3e164683eaf4631b989f8b4771f64e0c0972452d1555a15c65714b5ceefb295fb7e
+checksum=9e322782bc77c740ce956d57d0d89ec9ac9f2b890c474ac937f400db848df3e164683eaf4631b989f8b4771f64e0c0972452d1555a15c65714b5ceefb295fb7e
 format=raw
 size=4294967296
 compressed_size=149728112
@@ -154,7 +154,7 @@ name=Ubuntu 12.04 (Precise)
 osinfo=ubuntuprecise
 file=ubuntu-12.04.xz
 sig=ubuntu-12.04.xz.sig
-checksum[sha512]=15eab4610a44bf28423c08982cec47f6c3d43f29834791440249916ed76cc2dfeaf0405fddc8627086257bbbdaa4016c8982ad9b269f57f8f625b0e70d09e4e0
+checksum=15eab4610a44bf28423c08982cec47f6c3d43f29834791440249916ed76cc2dfeaf0405fddc8627086257bbbdaa4016c8982ad9b269f57f8f625b0e70d09e4e0
 format=raw
 size=4294967296
 compressed_size=168593316
@@ -178,7 +178,7 @@ name=Ubuntu 13.10 (Saucy)
 osinfo=ubuntusaucy
 file=ubuntu-13.10.xz
 sig=ubuntu-13.10.xz.sig
-checksum[sha512]=841876f027c63229a0aaac4540ba80d23e56d211138987e3d8187cacb04bc801d144a9831d842fc04aea557fc0c690c812c045ed440123b1d616e61ab2d56049
+checksum=841876f027c63229a0aaac4540ba80d23e56d211138987e3d8187cacb04bc801d144a9831d842fc04aea557fc0c690c812c045ed440123b1d616e61ab2d56049
 format=raw
 size=4294967296
 compressed_size=196481392
diff --git a/builder/website/index.asc b/builder/website/index.asc
index fdff3b2..087ebeb 100644
--- a/builder/website/index.asc
+++ b/builder/website/index.asc
@@ -7,7 +7,7 @@ revision=4
 osinfo=centos6.4
 file=centos-6.xz
 sig=centos-6.xz.sig
-checksum[sha512]=cc961bee19e0ed9573f2fa1bb26923bc77b882fe9b82eb0fefdf969237b586ebb6b4a02818321ace6cc3604d8f0f60e76f22039bf301093fa944c27399f70904
+checksum=cc961bee19e0ed9573f2fa1bb26923bc77b882fe9b82eb0fefdf969237b586ebb6b4a02818321ace6cc3604d8f0f60e76f22039bf301093fa944c27399f70904
 format=raw
 size=6442450944
 compressed_size=191253016
@@ -29,7 +29,7 @@ name=Debian 6 (Squeeze)
 osinfo=debian6
 file=debian-6.xz
 sig=debian-6.xz.sig
-checksum[sha512]=1baff51aae7cb89d96c5a95bcd18f3f9ce04945366802107a4ed3b5bd3f9c1fa77af8aa9716cea2826211ef03855a3f02dfa2097daeae66d4ec20a348c820354
+checksum=1baff51aae7cb89d96c5a95bcd18f3f9ce04945366802107a4ed3b5bd3f9c1fa77af8aa9716cea2826211ef03855a3f02dfa2097daeae66d4ec20a348c820354
 format=raw
 size=4294967296
 compressed_size=141531780
@@ -59,7 +59,7 @@ name=Debian 7 (Wheezy)
 osinfo=debian7
 file=debian-7.xz
 sig=debian-7.xz.sig
-checksum[sha512]=1c36eb94401acc567a1c542e4eafde2ba15dab6b615eb5a5bf4e18895c30d825e2e57c6d3a62f998dfc232f3747d5c92d2c5a497ac5d62d73426da5bf80652a5
+checksum=1c36eb94401acc567a1c542e4eafde2ba15dab6b615eb5a5bf4e18895c30d825e2e57c6d3a62f998dfc232f3747d5c92d2c5a497ac5d62d73426da5bf80652a5
 format=raw
 size=4294967296
 compressed_size=149230076
@@ -89,7 +89,7 @@ name=Fedora速 18
 osinfo=fedora18
 file=fedora-18.xz
 sig=fedora-18.xz.sig
-checksum[sha512]=12435775193b69f6e22658aaa001d4ca9b15fd68a04b4b7e9be20b3b517e857e417dc3268a302979d4a702b20f25754025f7ae0e9fb7088419a4ca1669585e6f
+checksum=12435775193b69f6e22658aaa001d4ca9b15fd68a04b4b7e9be20b3b517e857e417dc3268a302979d4a702b20f25754025f7ae0e9fb7088419a4ca1669585e6f
 format=raw
 size=6442450944
 compressed_size=148947524
@@ -111,7 +111,7 @@ name=Fedora速 19
 osinfo=fedora19
 file=fedora-19.xz
 sig=fedora-19.xz.sig
-checksum[sha512]=6228792b15df695260eea5530770e22b1bce40a1986410252928adbe4131ab85c031bcf8892736b63240fd585e39a5f44ed7770ba0dc1e6d8de6834cccee3767
+checksum=6228792b15df695260eea5530770e22b1bce40a1986410252928adbe4131ab85c031bcf8892736b63240fd585e39a5f44ed7770ba0dc1e6d8de6834cccee3767
 format=raw
 size=4294967296
 compressed_size=172190964
@@ -133,7 +133,7 @@ name=Ubuntu 10.04 (Lucid)
 osinfo=ubuntulucid
 file=ubuntu-10.04.xz
 sig=ubuntu-10.04.xz.sig
-checksum[sha512]=9e322782bc77c740ce956d57d0d89ec9ac9f2b890c474ac937f400db848df3e164683eaf4631b989f8b4771f64e0c0972452d1555a15c65714b5ceefb295fb7e
+checksum=9e322782bc77c740ce956d57d0d89ec9ac9f2b890c474ac937f400db848df3e164683eaf4631b989f8b4771f64e0c0972452d1555a15c65714b5ceefb295fb7e
 format=raw
 size=4294967296
 compressed_size=149728112
@@ -157,7 +157,7 @@ name=Ubuntu 12.04 (Precise)
 osinfo=ubuntuprecise
 file=ubuntu-12.04.xz
 sig=ubuntu-12.04.xz.sig
-checksum[sha512]=15eab4610a44bf28423c08982cec47f6c3d43f29834791440249916ed76cc2dfeaf0405fddc8627086257bbbdaa4016c8982ad9b269f57f8f625b0e70d09e4e0
+checksum=15eab4610a44bf28423c08982cec47f6c3d43f29834791440249916ed76cc2dfeaf0405fddc8627086257bbbdaa4016c8982ad9b269f57f8f625b0e70d09e4e0
 format=raw
 size=4294967296
 compressed_size=168593316
@@ -181,7 +181,7 @@ name=Ubuntu 13.10 (Saucy)
 osinfo=ubuntusaucy
 file=ubuntu-13.10.xz
 sig=ubuntu-13.10.xz.sig
-checksum[sha512]=841876f027c63229a0aaac4540ba80d23e56d211138987e3d8187cacb04bc801d144a9831d842fc04aea557fc0c690c812c045ed440123b1d616e61ab2d56049
+checksum=841876f027c63229a0aaac4540ba80d23e56d211138987e3d8187cacb04bc801d144a9831d842fc04aea557fc0c690c812c045ed440123b1d616e61ab2d56049
 format=raw
 size=4294967296
 compressed_size=196481392
@@ -202,17 +202,17 @@ notes=Ubuntu 13.10 (Saucy).
 -----BEGIN PGP SIGNATURE-----
 Version: GnuPG v1.4.14 (GNU/Linux)
 
-iQIcBAEBAgAGBQJScBfxAAoJEJFzj3Pht2igVKoP/RoXZQsFe7KWFUXnWNKlwexV
-QYuqoN/qhRO38JJVGdBlDFgBSgDlseSC69LSJKn4/Z3CkTUROE1tvt5K5ImYyL4w
-iwS9xgDb/pcnLIE8Z+mCxx5ozCTprYxbhCkYKaxG8bKs+qYlILk0DQTkXd/lTPuk
-GlePES/NJi6Ekp7cjtNnt3Wxpmweyd4CZvUamBWDLb7eLSsgtCWDR0Ble6x0XEXJ
-MGAxrOhrmQeUXa8eTep3z/PfVHeEwUE1vMsVRds5Ewo0w1XxmJhb0csXTmhvWUiJ
-Curx+krSEUvn+lpM9V5zP00rtyTWJL7M9lW85rtO5WXmYthHNK6xKvMV91rBjuCK
-4iO/ua7vZMCb2u4XoSZ4Xd/0EGJFOXGwE2xICGvzSg8NcvajreTQrAFH/QvUQ6Uv
-Jag7Jxeg+PlYoz5lHX7rWbE9gB4WUcQWVQfr3lMdpDc3/ZJk9WVjrTJ8haTC/fFA
-+guf1vTbcMRfxYRMtpXmmnvCz/FHLengTZYmxaM0z4cQF3tV4whj+uXS5n1pbmrE
-oBV9YBmYGmPQdQlmbL8/pIOnzdZu1M2FW3zP1+1ELpciSlCZ7pi8emSl/vMf+qvE
-/4v2ydeWnWdrev1ShKlOeq6cbNiGwP5k0Kq2lLAq90Co2mmlxmJxOfS45SxqyspD
-oK4BPpCitrXKfrf1tItA
-=B2fm
+iQIcBAEBAgAGBQJSc9qSAAoJEJFzj3Pht2igkYkQAKoYhPycj0ydM5Gk3nbaSOuU
+Rsz1LsujBsYTM+E51UjBm4bcCJw4HwHYRUMT/t5cdMSX/NjSNGHjeLRBQvOGJMHx
+DV6quFv/mwSCjvncWPuT3iCRNIIAq0ECr5YvfKhJ6HtoD/cQM7WUivYdZUJERC89
+H1d5bjDiU25qLrhl68fEMyQzzpKIq3CbsIsczlCxxuHefXNhhp7XmqGtrNwpd7im
+FtbKaIPeUDreoDCNNZzeUwdJ3TR6ybPB1Iy5Pl3kaBcM3VKqV5qXdQdmCbLyVtzg
+rg4cSxTRiMJGstMVIiDJzsS+DGfUngpXZnxndDpzaX6BVgMqFjSPcjjif2f+wWsK
+t1eXAhyoUYK+mepsNmYsOzBLFgnQKnmRO4cy6DgKq+twrx4v7snDpQ8J+HZkGHM6
+sdliAQ7QEBmcLVArlzsq61RP3XlWrSIwmRG7BBGFptom4p+m7JeIPCH9EKZ9QIDK
+tWOq8LqEAsNvaJrw+HTcGaryEYOf6YmJFd0iHgqDOnrsvsPK9Ex8hhC05jYe/Lkp
+HzZiRjHFpHbm8lBVFcrajyGumdIieg2EUh7iygxaRFQa+7UA7fkYuGhYXKcsbgyT
+Q3/c5TUAAM9qpj70rtTHV0LCmXL21aPqsMRPD/Oawe9N2iYe0HfNuZ1bBauPJEGy
+pixuD1Mt3Xl/2Hl0GVcY
+=x34e
 -----END PGP SIGNATURE-----
-- 
1.8.3.1

